app/controllers/posts_controller.rb

```diff
@@ -16,10 +18,30 @@ def new
   end
 
   def create
-    @user = current_user
-    @post = Post.new(author_id: @user, title: params[:post][:title], text: params[:post][:text])
-    @post.author_id = @user.id
-    @post.save
-    redirect_to user_posts_path(@user)
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+    return unless @post.save
+
+    redirect_to user_posts_path
+    flash[:success] = 'Post created!'
+  end
+
+  def destroy
+    @post = Post.find(params[:id])
+    if current_user.admin? || current_user == @post.user
+      @post.destroy
+      flash[:success] = 'Post deleted!'
+    else
+      flash[:error] = 'You are not authorized to delete this post.'
+    end
+    redirect_to user_posts_path
+  end
```

Comment:
- [ ] As Mentioned by the previous reviewer and for project requirements,  Kindly ensure that the delete post functionality is working for the admin role. Only the post author can delete a post. Right now the admin can only delete a comment. Below is the error I get when truing to delete a post for a user with an `admin` role.
![Screenshot from 2023-04-20 20-13-36](https://user-images.githubusercontent.com/83042742/233452515-3bfd4258-f8ce-4ba9-9ae2-eee22eb9be5d.png)